gpflow/base.py

```diff
@@ -59,17 +61,29 @@ def __init__(self,
             self._unconstrained = tf.Variable(value, dtype=dtype, name=name, trainable=trainable)
 
         self.prior = prior
+        self.prior_on_x = prior_on_x
         self._transform = transform
 
-    def log_prior(self):
+    def log_prior(self, evaluate_on_x: bool = False):
```

Comment:
```suggestion
    def log_prior(self, evaluate_in_constrained_space: bool = False):
```
(or something similar)